Source/JavaScriptCore/interpreter/Interpreter.cpp

```diff
@@ -122,10 +122,23 @@ JSValue eval(CallFrame* callFrame, JSValue thisValue, JSScope* callerScopeChain,
         return jsUndefined();
 
     JSValue program = callFrame->argument(0);
-    if (!program.isString())
-        return program;
+    JSString* programString = nullptr;
+    if (Options::useTrustedTypes()) {
+        if (LIKELY(program.isString()))
+            programString = asString(program);
+        else if (program.isObject()) {
+            auto code = globalObject->globalObjectMethodTable()->codeForEval(globalObject, program);
+            if (code.isNull())
+                return program;
+            programString = JSC::jsString(vm, code);
+        } else
+            return program;
+    } else {
+        if (!program.isString())
+            return program;
+        programString = asString(program);
+    }
```

Comment:
I think this is a bit cleaner as: 

```c++
JSString* programString = nullptr;
if (LIKELY(program.isString())
    programString = asString(program);
else if (Options::useTrustedTypes()) {
    auto code = globalObject->globalObjectMethodTable()->codeForEval(globalObject, program);
    if (!code.isNull())
        programString = jsString(vm, code);
}

if (!programString)
    return program;
```

But I was under the impression that you could have TrustedTypes be `eval`able and not allow string eval. This current code ties those together.

Reply:
The blocking of string eval is being done in a follow up to this, this just handles evaling trusted script objects.

That does indeed look cleaner I'll update the PR.